server/ec2spotmanager/cron.py

```diff
@@ -107,33 +109,32 @@ def check_instance_pools():
 
 @app.task
 def update_spot_prices():
+    from .models import EC2PoolConfiguration
+    logger = logging.getLogger("ec2spotmanager")
     """Periodically refresh spot price history and store it in redis to be consumed when spot instances are created.
 
     Prices are stored in redis, with keys like:
         'ec2spot:price:{instance-type}'
     and values as JSON objects {region: {az: [prices]}} like:
         '{"us-east-1": {"us-east-1a": [0.08000, ...]}}'
     """
-    from .common.prices import get_spot_prices
     from .models import PoolConfiguration
 
     regions = set()
     for cfg in PoolConfiguration.objects.all():
-        if cfg.ec2_allowed_regions:
-            regions |= set(json.loads(cfg.ec2_allowed_regions))
-
-    now = timezone.now()
-    expires = now + datetime.timedelta(hours=3)  # how long this data is valid (if not replaced)
-
-    prices = get_spot_prices(regions,
-                             getattr(settings, 'AWS_ACCESS_KEY_ID', None),
-                             getattr(settings, 'AWS_SECRET_ACCESS_KEY', None))
-
-    # use pipeline() so everything is in 1 transaction
-    cache = redis.StrictRedis(host=settings.REDIS_HOST, port=settings.REDIS_PORT, db=settings.REDIS_DB).pipeline()
-    for instance_type in prices:
-        key = 'ec2spot:price:' + instance_type
-        cache.delete(key)
-        cache.set(key, json.dumps(prices[instance_type], separators=(',', ':')))
-        cache.expireat(key, expires)
-    cache.execute()  # commit to redis
+        if cfg.provider.ec2_allowed_regions:
+            regions |= set(json.loads(cfg.provider.ec2_allowed_regions))
+
+        now = timezone.now()
+        expires = now + datetime.timedelta(hours=3)  # how long this data is valid (if not replaced)
+        cloud_provider = Provider().getInstance(config=cfg.provider)
+        if isinstance(cfg.provider, EC2PoolConfiguration):
```

Comment:
These two lines are using two different ideas for abstraction. You have the Provider interface and then model-based inheritance. Why use both?

Reply:
This has been removed. We are now just using the interface.